src/protocol.rs

```diff
@@ -285,9 +294,40 @@ impl ParticipantConfig {
             })
             .collect()
     }
+
+    fn random<R: RngCore + CryptoRng>(size: usize, rng: &mut R) -> ParticipantConfig {
+        let ids = std::iter::repeat_with(|| ParticipantIdentifier::random(rng))
+            .take(size)
+            .collect::<Vec<_>>();
+
+        let random_configs: Vec<ParticipantConfig> = (0..size)
+            .map(|i| {
+                let mut other_ids = ids.clone();
+                let id = other_ids.swap_remove(i);
+                Self { id, other_ids }
+            })
+            .collect();
```

Comment:
todo: this is a bit overkill -- it makes a whole set of configs when we really just need one! I would just generate the one additional `self` id, like
```rust
// normally we don't put asserts in code, but since this is for testing only, it's ok.
assert!(size > 1);
let other_ids = std::iter::repeat_with(|| PId::random(rng))
    .take(size - 1)
    .collect::<Vec<_>>();
let id = PId::random(rng);
Self { id, other_ids }
```